File: server/Freshmart.Infrastructure/Data/DbEntities/Order.cs
using System;
using System.Collections.Generic;

namespace Freshmart.Infrastructure.Data.DbEntities
{
    public class Order : BaseEntity
    {
        public Order()
        {
            OrderItems = new HashSet<OrderItem>();
        }
        
        public Guid UserId { get; set; }
        public string OrderNumber { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public decimal TotalAmount { get; set; }
        public string Status { get; set; } = "Pending";
        
        // Address details
        public string ShippingAddress { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string ZipCode { get; set; }
        public string Country { get; set; }
        public string PhoneNumber { get; set; }
        
        // Email for invoice
        public string Email { get; set; }
        
        // Navigation properties
        public virtual ApplicationUser User { get; set; }
        public virtual ICollection<OrderItem> OrderItems { get; set; }
    }
} 

File: client/src/types/order.ts
export interface OrderItem {
  id: string
  productId: string
  productName: string
  unitPrice: number
  quantity: number
  subtotal: number
  productImageUrl: string
}

export interface Order {
  id: string
  userId: string
  orderNumber: string
  orderDate: string
  totalAmount: number
  status: string
  shippingAddress: string
  city: string
  state: string
  zipCode: string
  country: string
  phoneNumber: string
  email: string
  userName: string
  items: OrderItem[]
}

export interface CreateOrderItem {
  productId: string
  quantity: number
}

export interface CreateOrderRequest {
  items: CreateOrderItem[]
  shippingAddress: string
  city: string
  state: string
  zipCode: string
  country: string
  phoneNumber: string
  email: string
}

export interface OrdersResponse {
  success: boolean
  message: string
  statusCode: number
  data: Order[]
  totalCount: number
  totalPages: number
  page: number
  pageSize: number
}

export interface OrderResponse {
  success: boolean
  message: string
  statusCode: number
  data: Order
}

export interface ShippingAddress {
  street: string
  city: string
  state: string
  zipCode: string
  country: string
  phoneNumber: string
  email: string
}
